src/userdev/java/net/minecraftforge/gradle/userdev/UserDevPlugin.java

```diff
@@ -293,6 +252,50 @@ public void apply(@Nonnull Project project) {
 
             extension.getRuns().forEach(runConfig -> runConfig.token("asset_index", finalAssetIndex));
             Utils.createRunConfigTasks(extension, extractNatives, downloadAssets, createSrgToMcp);
+
+            if (doingUpdate) {
+                logger.lifecycle("This process uses Srg2Source for java source file renaming. Please forward relevant bug reports to https://github.com/MinecraftForge/Srg2Source/issues.");
+
+                final String[] updateSourceSets = project.hasProperty("UPDATE_SOURCESETS") ? ((String) project.property("UPDATE_SOURCESETS")).split(";") : new String[] { SourceSet.MAIN_SOURCE_SET_NAME };
```

Comment:
Main thing of note when it comes to this is that two things must be true:
- If the property exists, all source sets wanting to be updated must be specified. This may just want to include main by default, but that more QoL than anything.
- The SourceSets listed must have no whitespace whether leading or trailing.

These two are mainly considerations, but neither I believe are necessary for change as long as those nuances are specified somewhere.

Reply:
Honestly this whole thing needs to be moved to a separate reusable script, considering its copy pasta in multiple of the plugins we expose.